### ISSUE TITLE
C-4 has no `need_wielding`, and can't have one

### ISSUE THREAD

**Issue by GuardianDll** [CleverRaven/Cataclysm-DDA#64308]
### Describe the bug

Which makes interesting issues, when people drop the item on the ground, activate it, and the item (instead of remaining on the ground) teleport right into player's pocket

### Attach save file

N/A

### Steps to reproduce

1. Spawn C-4
2. Drop it
3. `a`ctivate it when it's on the ground
4. The game allow you to activate it
5. After activation the item teleport into your pockets

### Expected behavior

Easiest: you can't activate C-4 remotely
Better: jsonify C-4, so it can use need_wielding
The best: jsonify C-4, but also add a boolean `remote_use`, which allow item to be activated in adjacent tiles, emulating "i put C-4 (or any another explosive) near the wall, and activate it here"

### Screenshots

![GIF 16-03-2023 17-57-56](https://user-images.githubusercontent.com/67688115/225707417-df67b111-a6d6-43ea-a8bc-52f09af6c401.gif)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2364 (21H2)
- Game Version: 8286f4d [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Magiclysm [magiclysm],
    Xedra Evolved [xedra_evolved]
]

**Comment by aahue1889** [comment]
Ack looking into

**Comment by aahue1889** [comment]
The Json change is going to be here:
https://github.com/CleverRaven/Cataclysm-DDA/blob/9777f4e4d2b29cc00eefc6d3514a133d8728ab33/data/json/items/tool/explosives.json
---

The "c4" code that is going to be Jsonified

![image](https://user-images.githubusercontent.com/8581536/225806606-afe62480-3672-4953-9285-06d2cef8dc01.png)

--

Going to continue to look at this

**Comment by aahue1889** [comment]
I didn't really have to do anything in the C++ code; all I did was just add FLAGS and now the C4 works as expected 
- Going to spend some time testing this tomorrow and confirm these other flags haven't messed with the behavior of C4

![image](https://user-images.githubusercontent.com/8581536/225820173-63c4d166-f79c-40b9-b142-22aa90dcc21a.png)

**Comment by GuardianDll** [comment]
ALLOWS_REMOTE_USE fox this, but also unlock the ability to activate c-4 right from your inventory, and i don't know can it cause some problems or not

**Comment by aahue1889** [comment]
Still testing; but here is a recording of the C4 working as Requested (no teleporting into pocket)

https://user-images.githubusercontent.com/8581536/225964275-9c285bdb-69b6-46eb-bb11-5fe3b9dfcafa.mp4

--- Still testing other Flags Radio_Invoke/ Radio_Modable/ Bomb

**Comment by aahue1889** [comment]
Radio_Modable and Radio_invoke_Proc both work
- If you use a RC Controller and attach 'Red' Signal using the "Radio Activation Mod" you will be prompted to set timer - and the C4 will remotely explode
- BOMB works as well; but you can only throw the c4 2 Square away... Not sure if this is something handy.
---

Going to commit this code after I add this feature to the Demo Charges as well; they both are 'stick here' then ignite bombs

```
{
    "id": "homemade_demolition_charge",
    "type": "TOOL",
    "category": "weapons",
    "name": { "str": "homemade demolition charge" },
    "description": "An improvised demolition charge.  Use this item to light the fuse.  You will, of course, need a lighter or matches in your inventory to do this.  20 seconds after lighting the fuse, it will explode, so get away!",
    "weight": "3875 g",
    "volume": "3900 ml",
    "price": 0,
    "to_hit": -3,
    "bashing": 4,
    "material": [ "plastic", "powder" ],
    "symbol": "*",
    "color": "red",
    "explode_in_fire": true,
    "explosion": { "power": 2590 },
    "use_action": {
      "need_wielding": true,
      "target": "homemade_demolition_charge_act",
      "msg": "You light the %s.",
      "target_charges": 20,
      "active": true,
      "need_fire": 1,
      "menu_text": "Light fuse",
      "type": "transform"
    },
    "flags": [ "RADIO_MODABLE", "RADIO_INVOKE_PROC", "BOMB", "ALLOWS_REMOTE_USE" ]
  },
```

**Comment by GuardianDll** [comment]
fixed in #64347
